dask/dataframe/io/parquet/core.py

```diff
@@ -1042,21 +1032,27 @@ def get_engine(engine):
 
         if engine in ("pyarrow", "arrow"):
             engine = "pyarrow-dataset"
-        elif pa_version.major >= 5 and engine == "pyarrow-legacy":
-            warnings.warn(
-                "`ArrowLegacyEngine` ('pyarrow-legacy') is deprecated for "
-                "pyarrow>=5 and will be removed in a future release. Please "
-                "use `engine='pyarrow'` or `engine='pyarrow-dataset'`.",
-                FutureWarning,
-            )
 
         if engine == "pyarrow-dataset":
+            if pa_version.major < 1:
```

Comment:
Can we not require >1 by now?

Reply:
We probably can since we dropped 3.7 support, but pyarrow-1 *should* still work with the pyarrow-dataset engine. I figured we could bump this later on when needed, but I'm open to anything

Reply:
What you have is fine. We'll simply remove the whole thing soon enough!